### PR TITLE
feat: split edge functions

### PR DESCRIPTION
**Pull request by teemingc**
This PR removes the arbitrary limitation preventing Netlify edge functions from being split. It also adds a variation of the `split` test suite with `edge: true` set

---

### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:

- [ ] It's really useful if your PR references an issue where it is discussed ahead of time. In many cases, features are absent for a reason. For large changes, please create an RFC: https://github.com/sveltejs/rfcs
- [x] This message body should clearly illustrate what problems it solves.
- [x] Ideally, include a test that fails without this PR but passes with it.

### Tests

- [ ] Run the tests with `pnpm test` and lint the project with `pnpm lint` and `pnpm check`

### Changesets

- [x] If your PR makes a change that should be noted in one or more packages' changelogs, generate a changeset by running `pnpm changeset` and following the prompts. Changesets that add features should be `minor` and those that fix bugs should be `patch`. Please prefix changeset messages with `feat:`, `fix:`, or `chore:`.

### Edits

- [x] Please ensure that 'Allow edits from maintainers' is checked. PRs without this option may be closed.

### CODE REVIEW

File: packages/adapter-netlify/index.js [hunk 1]
```diff
@@ -87,22 +80,23 @@ export default function ({ split = false, edge = edge_set_in_env_var } = {}) {
 			builder.writeClient(publish_dir);
 			builder.writePrerendered(publish_dir);
 
-			// Copy user's custom _headers file if it exists
 			if (existsSync('_headers')) {
 				builder.copy('_headers', join(publish, '_headers'));
 			}
 
+			if (existsSync('_redirects')) {
```
Comment (teemingc) [inline packages/adapter-netlify/index.js:98]: This was previously copied only when edge was false, which is incorrect. This PR fixes that by hoisting it out of `generate_serverless_functions`

File: packages/adapter-netlify/index.js [hunk 2]
```diff
@@ -10,13 +10,6 @@ import { parse } from 'smol-toml';
 import { build } from 'rolldown';
 import { matches, get_publish_directory, s } from './utils.js';
 
-/**
- * @typedef {{
- *   build?: { publish?: string }
- *   functions?: { node_bundler?: 'zisi' | 'esbuild' }
- * } & TomlTable} NetlifyConfig
```
Comment (teemingc) [inline packages/adapter-netlify/index.js:18]: We don't use this type anywhere so I removed it

File: packages/adapter-netlify/files/edge.js [hunk 3]
```diff
@@ -1,41 +1,47 @@
...
+	 * @type {string}
+	 */
+	let origin;
 
-const initialized = server.init({
-	env: Deno.env.toObject(),
-	read: async (file) => {
-		const url = `${origin}/${file}`;
-		const response = await fetch(url);
+	const initialized = server.init({
```
Comment (teemingc) [inline packages/adapter-netlify/files/edge.js:16]: wrapped the logic of this file so that it's similar to the `files/serverless.js` one where we export a higher order function

File: packages/adapter-netlify/index.js (hunk 1 above)
Comment (teemingc) [reply packages/adapter-netlify/index.js:98]: split this into https://github.com/sveltejs/kit/pull/17047

File: packages/adapter-netlify/index.js [hunk 4]
```diff
@@ -260,6 +175,83 @@ function write_frameworks_config({ builder }) {
...
+			} else {
+				parts.push(segment.content);
+			}
+		}
+
+		// Netlify handles trailing slashes for us, so we don't need to include them in the pattern
+		const pattern = `/${parts.join('/')}`;
+
+		const name =
+			FUNCTION_PREFIX + (parts.join('-').replace(/[:.]/g, '_').replace(/\*/g, '__rest') || 'index');
```
[the remark this answers is not in the document]
Comment (teemingc) [reply packages/adapter-netlify/index.js]: The different pattern issue is being fixed in https://github.com/sveltejs/kit/pull/17049

File: .github/actions/netlify-deploy/action.yml [hunk 5]
```diff
@@ -32,6 +35,24 @@ runs:
...
+          --functions=".netlify/v1/functions" \
+          --auth="$NETLIFY_AUTH_TOKEN" \
+          --site="$NETLIFY_PROJECT_ID"
+        ); then
+          deploy_url=$(jq -er '.deploy_url' <<< "$deploy_json")
+          echo "url=$deploy_url" >> "$GITHUB_OUTPUT"
+        else
+          echo "::error::Netlify deployment failed"
+          echo "$deploy_json" >&2
+          exit 1
```
Comment (teemingc) [inline .github/actions/netlify-deploy/action.yml]: Helps error when deployment fails rather than silently passing

File: packages/adapter-netlify/index.js [hunk 6]
```diff
@@ -442,92 +444,227 @@ const rolldown_config = {
...
+		return create_function_config(
+			'edge',
+			fn.name,
+			fn.patterns,
+			excluded_paths.concat(fn.exclude ?? []),
+			fn.name
+		);
+	});
+
+	// always strip __pathname query parameter as it's reserved for split function reroutes
```
Comment (teemingc) [inline packages/adapter-netlify/index.js]: need to manually verify this

File: packages/adapter-netlify/test/apps/split/package.json [hunk 7]
```diff
@@ -6,7 +6,9 @@
 		"dev": "vite dev",
 		"build": "vite build",
 		"preview": "vite preview",
-		"test:build": "pnpm build && vitest run build.test.js",
+		"test:serverless:build": "pnpm build && vitest run build.test.js",
+		"test:edge:build": "EDGE=1 pnpm build && vitest run build.test.js",
```
Comment (teemingc) [inline packages/adapter-netlify/test/apps/split/package.json]:
```suggestion
		"test:edge:build": "EDGE=true pnpm build && vitest run build.test.js",
```

File: packages/adapter-netlify/index.js (hunk 6 above)
Comment (teemingc) [reply packages/adapter-netlify/index.js]: verified

File: .github/actions/netlify-deploy/action.yml (hunk 5 above)
Comment (teemingc) [reply .github/actions/netlify-deploy/action.yml]: split into https://github.com/sveltejs/kit/pull/17057

**Review by Rich-Harris — APPROVED** [review]
